### PR TITLE
q1

### PR DESCRIPTION
**Pull request by kobybs**

### CODE REVIEW

File: hellomod.c [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+int init_module(void)
+{
+    printk("Hello World\n");
+    return 0;
+}
+
+void cleanup_module(void)
+{       
+    printk("Module Removed\n");
+}
```
Comment (maykog) [inline hellomod.c]: Add a newline at the of the file

File: hellomod.c [hunk 2]
```diff
@@ -0,0 +1,17 @@
+#define MODULE
```
Comment (maykog) [inline hellomod.c:1]: Do you understand what the first 3 lines mean?

Comment (kobybs) [reply hellomod.c:1]: Defines can alter the behavior of the code by emitting or adding lines at the preprocessing phase.
for example at linux\init.h the exit behaviour is different for modules.

File: hellomod.c (hunk 1 above)
Comment (kobybs) [reply hellomod.c]: done

File: hellomod.c (hunk 2 above)
Comment (maykog) [reply hellomod.c:1]: I mean why do you need to define `MODULE`, `LINUX` and `__KERNEL__`?

**Comment by maykog** [comment]
you don't need to separate adding the newline into another commit, use `git commit --fixup`.

In addition, write more informative commit message. Rather than something like `q1`, you could write something like `hello module: Add basic module structure`.

File: hellomod.c (hunk 2 above)
Comment (kobybs) [reply hellomod.c:1]: __KERNEL__ - in header files which are used both by userspace and by kernel, __KERNEL__ is used to hide some declarations so they won't be seen by userspace code.
MODULE - used for dynamically loaded modules (as opposed to built-in drivers), to declare different entry and exit points.
LINUX - I grepped the entire src code but found no if/ndefs that mention LINUX so I guess it has no code related functionallity.

**Pull request closed without merging**